dev-packages/application-manager/src/generator/frontend-generator.ts

```diff
@@ -183,19 +183,36 @@ if (isMaster) {
                 shell.openExternal(url);
             });
 
-            const saveState = () => {
-                const bounds = newWindow.getBounds();
-                electronStore.set(windowStateName, {
-                    isMaximized: newWindow.isMaximized(),
-                    width: bounds.width,
-                    height: bounds.height,
-                    x: bounds.x,
-                    y: bounds.y
-                })
-            }
-            newWindow.on('close', saveState);
-            newWindow.on('resize', saveState);
-            newWindow.on('move', saveState);
+            // Save the window geometry state on every change
+            const saveWindowState = () => {
+                try {
+                    let bounds;
+                    if (newWindow.isMaximized()) {
+                        bounds = electronStore.get(WINDOW_STATE, {});
+                    } else {
+                        bounds = newWindow.getBounds();
+                    }
+                    electronStore.set(WINDOW_STATE, {
+                        isMaximized: newWindow.isMaximized(),
+                        width: bounds.width,
+                        height: bounds.height,
+                        x: bounds.x,
+                        y: bounds.y
+                    });
+                } catch (e) {
+                    console.error("Error while saving window state.", e);
+                }
+            };
+            let delayedSaveTimeout;
+            const saveWindowStateDelayed = () => {
+                if (delayedSaveTimeout) {
+                    clearTimeout(delayedSaveTimeout);
+                }
```

Comment:
Nit: You don't need to check for `if (delayedSaveTimeout)`, because [Passing an invalid ID to `clearTimeout()` silently does nothing; no exception is thrown.](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/clearTimeout#Notes)

Or, if you really want to implement an `if`, please fix the logic by setting `delayedSaveTimeout = undefined` after clearing the timeout.

Reply:
The link you provided points to the web API. Does it also hold for the Node.js API?